notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,330 @@
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_20_0.png)
+
+
+This process can go on and on, showing that when we can use any scalar multiples, we can make an infinite number of linear combinations. However, infinite does not mean any--the linear combinations span a space defined by the original vectors. In this case, we would create a plane as you can start to see above. Formally, this illustrates how vectors $\vec{x}$ and $\vec{1}$ span a vector space.
+
+Recall:
+- our scatter plot
+- $X$ and $\vec{y}$ are defined by the data
+- our loss value is the L2 norm of $y-X \hat{\theta}$, squared. 
+
+By inspection, we see that no line can perfectly fit our points, so we cannot achieve 0 loss. Thus, we know that $\vec{y}$ is not in the plane spanned by $\vec{x}$ and $\vec{1}$.
+
+Visually,
+
+<img src="../../notebooks-images/linear_projection1.png" width="500" />
+
+Since our error is based on distance, we can see that to minimize $ L(\hat{\theta}, \vec{x}, \vec{y}) = \left \Vert  \vec{y} - X \hat{\theta} \right \Vert ^2$, we want $X \hat{\theta}$ to be as close to $\vec{y}$ as possible.
+
+Mathematically, we are looking for the projection of $\vec{y}$ onto the vector space spanned by the columns of $X$ because the projection $\hat{y}$ has the property of being the closest point in the vector space to the vector $y$.
+<img src="../../notebooks-images/linear_projection2.png" width="500" />
+
+To see why, consider other points on the vector space, in purple.
+<img src="../../notebooks-images/linear_projection3.png" width="500" />
+
+By the Pythagorean Theorem, any other point on the plane is farther from $\vec{y}$ than $\hat{y}$ is. The longer the distance, the higher the least squared error. Thus, the length of the perpendicular represents the least squared error.
+
+## Linear Algebra
+
+Since we've snuck in a lot of linear algebra concepts already, all that's left is solving for the $\hat{\theta}$ that yields our desired $\hat{y}$.
+
+A couple things to note: 
+
+<img src="../../notebooks-images/linear_projection5.png" width="500" />
+
+- $\hat{y} + e = \vec{y}$
+- $\vec{e}$ is perpendicular to $\vec{x}$ and $\vec{1}$
+- $\hat{y} = X \hat{\theta^*}$ is the vector closest to $y$ in the vector space spanned by $\vec{x}$ and $\vec{1}$
+
+Thus, we arrive at the equation:
+
+$$X  \hat{\theta^*} + \vec{e} = \vec{y}$$
+
+Left-multiplying both sides by $X^T$:
+
+$$X^T X  \hat{\theta^*} + X^T \vec{e} = X^T \vec{y}$$
+
+But since $\vec{e}$ is perpendicular to the columns of $X$, $X^T \vec{e}$ is a column vector of $0$'s. Thus, we arrive at the Normal Equation:
+
+$$X^T X  \hat{\theta^*} = X^T \vec{y}$$
+
+From here, we can easily solve for $\hat{\theta^*}$ by left-multiplying both sides by $(X^T X)^{-1}$:
+
+$$\hat{\theta^*} = (X^T X)^{-1} X^T \vec{y}$$
+
+## Finishing up the Case Study
+
+Let's return to our case study and apply what we've learned, and explain to our client why our solution is sound.
```

Comment:
Hahahaha... we got rid of the client, remember?

Change to "and explain why our solution is sound"